tests/nat/nat_helpers.py

```diff
@@ -0,0 +1,848 @@
+POOL_RANGE_END_PORT = 6000
+logger = logging.getLogger(__name__)
+NAT_ADMIN_MODE = "enabled"
+STATIC_NAT_TABLE_NAME = "STATIC_NAT"
+STATIC_NAPT_TABLE_NAME = "STATIC_NAPT"
+DYNAMIC_POOL_NAME = "test_pool"
+ACL_TABLE_GLOBAL_NAME = "test_acl_table"
+DYNAMIC_BINDING_NAME = "test_binding"
+SUPPORTED_TOPO = ['t0', 't0-16', 't0-52', 't0-56', 't0-64', 't0-64-32', 't0-116']
+DIRECTION_MAP = {'leaf-tor': 'dnat'}
+ACL_SUBNET = "192.168.0.0/24"
+VRF = {"red": {"ip": "11.1.0.2", "id": "1", "mask": "30", "gw": "11.1.0.1", "port_id": "28", "dut_iface": "PortChannel0001"},
+       "blue": {"ip": "192.168.0.101", "id": "2", "mask": "24", "gw": "192.168.0.1", "port_id": "2"},
+       "green": {"ip": "12.1.0.2", "id": "3", "mask": "30", "gw": "12.1.0.1", "port_id": "29", "dut_iface": "PortChannel0002"},
+       "yellow": {"ip": "192.168.0.201", "id": "4", "mask": "24", "gw": "192.168.0.1", "port_id": "3"}
+       }
+SETUP_CONF = {"loopback": {"vrf": VRF, "acl_subnet": ACL_SUBNET},
+              "port_in_lag": {"vrf": VRF, "acl_subnet": ACL_SUBNET}
+              }
+DIRECTION_PARAMS = ['host-tor', 'leaf-tor']
+FULL_CONE_TEST_IP = "172.20.1.2"
+FULL_CONE_TEST_SUBNET = "172.20.1.0/24"
+REBOOT_MAP = {'cold': {"timeout": 300}, 'fast': {"timeout": 180}, 'warm': {"timeout": 180}}
+
+
+def configure_nat_over_cli(duthost, flow_direction, action, nat_type, global_ip, local_ip, proto=None,
+                           global_port=None, local_port=None):
+    """
+    static NAT/NAPT CLI wrapper
+    :param duthost: DUT host object
+    :param flow_direction: string traffic direction
+    :param action: string rule action
+    :param nat_type: string static nat type
+    :param global_ip: string global IP address value
+    :param local_ip: string local IP address value
+    :param proto: string protocol type
+    :param global_port: string global l4 port
+    :param local_port: string local l4 port
+    :return : dict with rule parameters
+    """
+    if nat_type == 'static_nat':
+        entry = {
+            global_ip: {'local_ip': local_ip}
+        }
+        duthost.command("sudo config nat {} static basic {} {}".format(action, global_ip, local_ip))
+        return entry
+    elif nat_type == 'static_napt':
+        static_napt_type = DIRECTION_MAP[flow_direction]
+        action_type_map = {'add': '-nat_type {}'.format(static_napt_type), 'remove': ''}
+        entry = {
+            "{}|{}|{}".format(global_ip, proto.upper(), global_port): {'local_ip': local_ip,
+                                                                       'local_port': "{}".format(local_port),
+                                                                       'nat_type': static_napt_type
+                                                                       }
+        }
+        duthost.command("sudo config nat {} static {} {} {} {} {} {}".format(action, proto.lower(),
+                                                                             global_ip, global_port,
```

Comment:
add a check  to see if the command is successful or not ?

Reply:
duthost.command will raise an error in case or return code != 0, as this is CLI command. In case of rc != 0 test will fail and teardown performed